hw7: phpunit

irbis45 opened the pull request:
Оптимизация кода
 Тесты для контроллеров
 Тесты для моделей
 Тесты для сервисов
 Тесты для репозиториев

egerasimchuk commented on online-consultant/app/Models/Company.php:
```diff
@@ -42,36 +42,87 @@
+     *
+     * @return mixed|string
+     */
+    public function getNameLinkAttribute()
```
Лучше логику отображения не вносить в модели, а вынести в хелперы

apavliukov replied:
То есть для каждой модели будет свой хелпер? Мне понравилась идея использовать $model->name_link, если будет ModelHelper::nameLink($model), то уже не так красиво. Хотя конечно захламлять модели этими аксессорами и правда не стоит. Может тогда вынести это в трейт, чтобы не терять возможность $model->name_link?

apavliukov commented on online-consultant/resources/views/admin/models/companies/forms/create.blade.php:
```diff
@@ -0,0 +1,37 @@
+{{ Form::open(['route' => 'admin.companies.store', 'class' => 'form-model']) }}
+    <div class="form-row">
```
Не заметил после всех правок, что разметка для create и edit стала одинаковой, но сейчас уже нет, в отдельный файл уже не вынести.

apavliukov commented on online-consultant/resources/views/admin/models/companies/forms/create.blade.php:
```diff
@@ -0,0 +1,37 @@
+            @include('common.forms.error', ['field' => 'url'])
+        </div>
+    </div>
+    @foreach($addressFields as $fieldName => $fieldLabel)
```
Эти поля можно назвать динамическими - скажем, в зависимости от страны юзера можно показывать/скрывать поле State. Чтобы не делать много отдельных столбцов в таблице БД, я решил эти поля хранить в одном столбце как массив и генерить из модели для views. Но где-то их хранить надо, где, если не в модели? То есть в каком одном месте и цеплять для разных views.

apavliukov commented on online-consultant/resources/views/admin/models/conversations/lists/index.blade.php:
```diff
@@ -0,0 +1,35 @@
+        <tbody>
+            @php /** @var \App\Models\Conversation $conversation */ @endphp
+            @foreach($conversations as $conversation)
+                <tr>
```
Думал об этом, смотря на ваш пример из урока. Но в моем случае компоненты таблицы неразделимы: этот <thead> принадлежит именно этой таблице и этой модели, тоже самое с <tr>. По мне так стоит выделять в отдельные views _общий_ код, который можно будет использовать в другом месте. А тут так не получится.

irbis45 committed:
hw7: phpunit
# Оптимизация кода
# Тесты для контроллеров
# Тесты для моделей
# Тесты для сервисов
# Тесты для репозиториев

irbis45 committed:
hw7
# Добавил тестов

irbis45 committed:
hw7
# Добавил в отслеживание еще файлы тестов
# рефакторинг

irbis45 committed:
hw7
# тесты для Fetcher

irbis45 committed:
hw7
# добавил тесты для updateController

irbis45 committed:
hw7
# добавил resources/sass в репозиторий
